src/pcp.rs

```diff
@@ -412,16 +412,17 @@ struct ProveShimGadget<F: FieldElement> {
 }
 
 impl<F: FieldElement> ProveShimGadget<F> {
+    #[allow(clippy::new_ret_no_self)]
     fn new(
         inner: Box<dyn Gadget<F>>,
         g_calls: usize,
     ) -> Result<Box<dyn Gadget<F>>, getrandom::Error> {
         let mut f_vals = vec![vec![F::zero(); 1 + g_calls]; inner.arity()];
         let mut prng = Prng::new_with_length(f_vals.len())?;
 
-        for wire in 0..f_vals.len() {
+        for wire in &mut f_vals {
             // Choose a random field element as the first point on the wire polynomial.
-            f_vals[wire][0] = prng.next().unwrap();
+            wire[0] = prng.next().unwrap();
```

Comment:
I don't like this change. `wire` is a good name for an index into the `f_vals`, but not a good name for a pointer to one of the values.

Reply:
I was on the fence about a lot of these, hence all the cases where I silenced the linter instead of introducing an iterator. I think I agree with you: the use of iterators is presumably natural for C++ programmers but I find them awkward coming from C, and agree that using indices is clearer for this usage. I'll flip this and the others you flagged back to the previous form and silence clippy where necessary.